src/config/compiler.rs

```diff
@@ -1,9 +1,45 @@
-use indexmap::IndexSet;
-
 use super::{
     builder::ConfigBuilder, graph::Graph, id::Inputs, transform::get_transform_output_ids,
     validation, Config, OutputId,
 };
+use std::fs::File;
+use std::path::PathBuf;
+
+use indexmap::IndexSet;
+
+#[cfg(not(test))]
+use fs4::FileExt;
+
+#[cfg(not(test))]
+fn create_data_dir_lock(data_dir: &Option<PathBuf>, errors: &mut Vec<String>) -> Option<File> {
```

Comment:
nit: Should this return a result instead of mutating a parameter? I think it just moves the `errors.push` down into the single caller.
```suggestion
fn create_data_dir_lock(data_dir: &Option<PathBuf>) -> Result<Option<File>, String> {
```

Reply:
Good point, thanks. Done in https://github.com/vectordotdev/vector/pull/19595/commits/015c2b53d38c6bc123bf3f30b34bffedb7671e08